hazelcast/src/main/java/com/hazelcast/core/IFunction.java

```diff
@@ -27,6 +29,8 @@
  * @param <R> a result
  * @since 3.2
  */
+
+@ClientProtocol
 public interface IFunction<T, R> extends Serializable {
 
```

Comment:
I was wondering whether we should annotate interfaces, I'm not sure myself.
Maybe it's good to do so, since the user will know that they should not chance their implementations.
On the other hand the interface does not have any serialization logic. 
But still it's probably better to annotate them.